### PR TITLE
[3.11] prevent external hostname pullthrough + bump digest cache size

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

File: pkg/dockerregistry/server/app.go [hunk 1]
```diff
@@ -23,8 +23,8 @@ import (
 
 const (
 	// Default values
-	defaultDescriptorCacheSize         = 4096
-	defaultDigestToRepositoryCacheSize = 2048
+	defaultDescriptorCacheSize         = 6 * 4096
+	defaultDigestToRepositoryCacheSize = 6 * 2048
```
Comment (smarterclayton) [inline pkg/dockerregistry/server/app.go]: I don't think you have to bump this, this is per digest anyway.

File: pkg/dockerregistry/server/app.go [hunk 2]
```diff
@@ -23,8 +23,8 @@ import (
 
 const (
 	// Default values
-	defaultDescriptorCacheSize         = 4096
-	defaultDigestToRepositoryCacheSize = 2048
+	defaultDescriptorCacheSize         = 6 * 4096
```
Comment (smarterclayton) [inline pkg/dockerregistry/server/app.go]: Add a comment about the rough size estimate ~16Mb of cache

File: pkg/imagestream/imagestream.go [hunk 3]
```diff
@@ -267,21 +267,31 @@ func (is *imageStream) Exists(ctx context.Context) (bool, *rerrors.Error) {
...
+	var localNames []string
+
 	local, err := imageapi.ParseDockerImageReference(stream.Status.DockerImageRepository)
 	if err != nil {
-		return "", rerrors.NewError(
-			ErrImageStreamUnknownErrorCode,
-			fmt.Sprintf("localRegistry: unable to parse reference %q", stream.Status.DockerImageRepository),
-			err,
-		)
+		glog.Warningf("localRegistry: unable to parse dockerImageRepository %q", stream.Status.DockerImageRepository)
```
Comment (dmage) [inline pkg/imagestream/imagestream.go]: We don't use glog, we use `dcontext.GetLogger(ctx).Warnf`

Comment (bparees) [reply pkg/imagestream/imagestream.go]: thanks, fixed.

**Pull request merged**